index.js

```diff
@@ -1,38 +1,46 @@
 'use strict'
 
 const fp = require('fastify-plugin')
-const amqpClient = require('amqplib/callback_api')
+const amqpClient = require('amqplib')
 
-function fastifyAmqp (fastify, opts, next) {
-  const host = opts.host
-
-  if (!host) {
-    next(new Error('`host` parameter is mandatory'))
-    return
+function getTarget ({
+  frameMax,
+  heartbeat,
+  hostname,
+  locale,
+  password,
+  port,
+  url,
+  username,
+  vhost
+}) {
```

Comment:
We should set default values to these options

Reply:
I was thinking to leave `amqplib` defaults (they are identical to the previous ones).